frame/balances/src/lib.rs

```diff
@@ -961,6 +965,7 @@ where
 		value: Self::Balance,
 		existence_requirement: ExistenceRequirement,
 	) -> DispatchResult {
+		if value.is_zero() { return Ok(()) }
```

Comment:
One side effect of this PR is that a `transfer` of zero just returns early.

Before, a transfer of zero would additionally charge a fee of `T::CreationFee` or `T::TransferFee` on top of the existing fee system.

Now, if the transfer is zero, because no logic is run, these additional fees are skipped. A user would still pay the normal transaction fee calculated with the `transaction_payment` module.

This all makes sense to me, but introduces a new behavior.

Reply:
It also means a transfer of 0 emits no `Balances` events.